### PR TITLE
RSDK-5693 fix tmc stepper motor index bug

### PR DESCRIPTION
**Pull request by oliviamiller**
- the tmc stepper motor indices were swapped, with 2 controlling the motor plugged into the channel labeled motor 1 and vice versa
- ideally I would want to change the `index` config field to `motor_channel` to match the other motor drivers but thats breaking so will leave as is.

TODO: 
 -will test with hardware when im back in office
-might want a migration script to switch the index values? otherwise users will have to manually switch them. not sure how many tmc motors are in prod.

### CODE REVIEW

File: components/motor/tmcstepper/stepper_motor_tmc.go [hunk 1]
```diff
@@ -112,7 +115,7 @@ const (
 	uSteps  = 256      // Microsteps per fullstep
 )
 
-// TMC5072 Register Addressses (for motor index 0)
+// TMC5072 Register Addressses (for motor index 1)
```
Comment (randhid) [inline components/motor/tmcstepper/stepper_motor_tmc.go:118]: Can't comment on the line, but could we unexport `NewMotor` on line 154, while we're at it?

**Review by randhid — APPROVED** [review]
There are a few optional add-ons to this PR for clean-up of constructor functions, but otherwise, LGTM!

**Comment by randhid** [comment]
@oliviamiller should we close this?

**Comment by oliviamiller** [comment]
> @oliviamiller should we close this?

yeah these changes were merged with [RSDK-6219](https://github.com/viamrobotics/rdk/pull/3929), closing

**Pull request closed without merging**